fhevm/tee_interpreter.go

```diff
@@ -405,57 +405,33 @@ func marshalTfheType(value any, typ tfhe.FheUintType) ([]byte, error) {
 func unmarshalTfheType(data []byte, typ tfhe.FheUintType) (uint64, error) {
 	switch typ {
 	case tfhe.FheBool:
-		if len(data) < 1 {
-			return 0, fmt.Errorf("data too short for type %s", typ)
-		}
-		if data[0] == 0 {
+		if data[len(data)-1] == 0 {
```

Comment:
Not related this this line, but putting here:

in tee_cast.go:
```
	if castToType == tfhe.FheUint4 {
		value = value % 16
	}
```

These 3 lines should go inside `marshalTfheType` function

Reply:
done